proxy/proxy/broker/rhnBroker.py

```diff
@@ -156,12 +158,20 @@ def _initConnectionVariables(self, req):
         # The auth token is sent in either a header or in the query part of the URI:
         # SLE minions -> query part of the URI.
         # RHEL minions -> 'X-Mgr-Auth' header.
+        # Debian -> Authorization (Basic Auth)
         #
         # Traditional SLE and RHEL clients uses 'X-RHN-Auth' header, but
         # no auth token is used in order to authenticate.
         if 'X-Mgr-Auth' in self.req.headers_in:
             self.authToken = self.req.headers_in['X-Mgr-Auth']
             del self.req.headers_in['X-Mgr-Auth']
+        elif 'Authorization' in self.req.headers_in and effectiveURI_parts.path.startswith('/rhn/manager/download/'):
+            # we need to remove Basic Auth, otherwise squid does not cache the package
+            # so we convert it into token auth
+            # The token is the login. So it is not secret
+            lpw = ustr(base64.b64decode(self.req.headers_in['Authorization'][6:])) # "Basic " == 6 characters
```

Comment:
@mcalmer I think it's quite possible to get exception here on calling `base64.b64decode`. Is it safe not to catch the exception here?

Reply:
I catch now exceptions, log an error and keep the Authorization header. It will fail later as it does now.